mne/channels/montage.py

```diff
@@ -1292,40 +1345,49 @@ def _set_montage(info, montage, update_ch_names=False, set_dig=True):
                  'left untouched.')
 
     elif isinstance(montage, DigMontage):
+        _mnt = montage if montage._coord_frame == 'head' else transform_to_head(montage.copy())
```

Comment:
don't rely on this private attribute _coord_frame that is meant to be removed. What defines the coord_frame is the coordinate system of the fiducials if any or if not present the coordinate system of all the points.